spec/javascripts/components/single-page-notification-button-spec.js

```diff
@@ -0,0 +1,102 @@
+/* eslint-env jasmine */
+/* global GOVUK */
+
+describe('Single page notification component', function () {
+  var FIXTURE
+
+  beforeEach(function () {
+    FIXTURE =
+      '<form class="gem-c-single-page-notification-button old-button-for-test" action="/email/subscriptions/single-page/new" method="POST" data-module="single-page-notification-button">' +
+        '<input type="hidden" name="base_path" value="/current-page-path">' +
+        '<button class="gem-c-single-page-notification-button__submit" type="submit">Get emails about this page</button>' +
+    '</form>'
+    window.setFixtures(FIXTURE)
+    jasmine.Ajax.install()
+  })
+
+  afterEach(function () {
+    jasmine.Ajax.uninstall()
+  })
+
+  it('calls the personalisation API on load', function () {
+    initButton()
+    var request = jasmine.Ajax.requests.mostRecent()
+    expect(request.url).toBe('/api/personalisation/check-email-subscription?base_path=/current-page-path')
+    expect(request.method).toBe('GET')
+  })
+
+  it('includes button_location in the call to the personalisation API when button_location is specified', function () {
+    FIXTURE =
+      '<form class="gem-c-single-page-notification-button old-button-for-test" action="/email/subscriptions/single-page/new" method="POST" data-module="single-page-notification-button" data-button-location="top">' +
+        '<input type="hidden" name="base_path" value="/current-page-path">' +
+        '<button class="gem-c-single-page-notification-button__submit" type="submit">Get emails about this page</button>' +
+      '</form>'
+    window.setFixtures(FIXTURE)
+
+    initButton()
+    var request = jasmine.Ajax.requests.mostRecent()
+    expect(request.url).toBe('/api/personalisation/check-email-subscription?base_path=/current-page-path&button_location=top')
+    expect(request.method).toBe('GET')
+  })
+
+  it('replaces the button when the API returns button html', function () {
+    initButton()
+
+    jasmine.Ajax.requests.mostRecent().respondWith({
+      status: 200,
+      contentType: 'application/json',
+      responseText: '{\n    "base_path": "/current-page-path",\n    "active": true,\n    "button_html": "<form class=\\"gem-c-single-page-notification-button new-button-for-test\\" action=\\"/email/subscriptions/single-page/new\\" method=\\"POST\\">\\n  <input type=\\"hidden\\" name=\\"base_path\\" value=\\"/current-page-path\\">\\n  <button class=\\"gem-c-single-page-notification-button__submit\\" type=\\"submit\\">Stop getting emails about this page\\n</button>\\n</form>"\n}'
+    })
+
+    var form = document.querySelector('form.gem-c-single-page-notification-button.new-button-for-test')
+    expect(form).toHaveText('Stop getting emails about this page')
```

Comment:
I wonder if this test needs to be more rigorous - should it check that the form contains a button and is therefore functional?

Reply:
sure, that is a fair point. The form doesn't do anything without a button after all. Will update